pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1469,8 +1472,34 @@ impl<'a> DatadirModification<'a> {
         content: &[u8],
         ctx: &RequestContext,
     ) -> anyhow::Result<()> {
-        let policy = self.tline.get_switch_aux_file_policy();
-        if let AuxFilePolicy::V2 | AuxFilePolicy::CrossValidation = policy {
+        let switch_policy = self.tline.get_switch_aux_file_policy();
+
+        let policy = {
+            let current_policy = self.tline.last_aux_file_policy.load();
+            // Allowed switch path:
+            // * no aux files -> v1/v2/cross-validation
+            // * cross-validation->v2
+            match (current_policy, switch_policy) {
+                (RuntimeAuxFilePolicy::Unspecified, _)
+                | (RuntimeAuxFilePolicy::CrossValidation, SwitchAuxFilePolicy::V2) => {
+                    let new_policy = switch_policy.to_runtime_policy();
+                    self.tline.last_aux_file_policy.store(new_policy);
+                    info!("switching to aux file policy {:?}", new_policy);
+                    if let Some(ref remote_client) = self.tline.remote_client {
+                        remote_client
+                            .schedule_index_upload_for_aux_file_policy_update(new_policy)?;
+                        remote_client.wait_completion().await?;
```

Comment:
I was earlier worried about the consistency if a branch is created without the index_part.json having persisted, but some writes having already occurred. I understood the clarification to be that branches have separate aux files, so it doesn't matter if the value is reset.

This `wait_completion` here will stall ingestion until we complete the upload, which is usually very fast, but there are some reasons for it to be slow and that could be other tenants.

If the child branch doesn't need to be consistent with parent branch aux files, why do we need this?